Rmarkdown/Multiple_Ascending_Dose_PKPD_continuous.Rmd

```diff
@@ -13,70 +13,88 @@ This document contains exploratory plots for continuous PD data as well as the R
-# version matches package in first filepath, in the case of multiple instances of a package
-
-# library(rmarkdown)
-library(gridExtra)
-library(grid)
+```{r, warning=FALSE, message=FALSE}
 library(ggplot2)
 library(dplyr)
-library(RxODE)
-library(caTools)
-
-theme_set(theme_bw(base_size=12))
+library(tidyr)
+library(xgxr, lib.loc = "../Rlib")
 
 #flag for labeling figures as draft
-  draft.flag      = TRUE
+status = "DRAFT"
 
 ## ggplot settings
-theme_set(theme_bw(base_size=12))
-
-```
-
-
-## Define Useful Graphics Functions
-
-```{r}
-
-# annotation of plots with status of code
-# add draft to plots:draft.flag = TRUE or FALSE - indicates whether to add draft
-AnnotateStatus <- function(draft.flag, x.pos=Inf, y.pos=Inf,
-                           label="DRAFT",color="grey",
-                           hjust=1.2,vjust=1.2,
-                           fontsize=7,alpha = 0.7,
-                           fontface="bold") {
-  
-    if(draft.flag) {
-    annotateStatus <- annotate("text",x=x.pos,y=y.pos,
-             label=label,
-             color=color,
-             hjust=hjust,vjust=vjust,
-             cex=fontsize, alpha = alpha,fontface=fontface)
-    } else {
-    annotateStatus <- NULL
-  }
-    return(annotateStatus)
-  }  
-
+xgx_theme_set()
 
 ```
 
 ## Load Dataset
 
 ```{r, warning=FALSE, message=FALSE}
-my.data <- read.csv("../Data/Multiple_Ascending_Dose_Dataset2.csv")
-
-# Define order for factors
-my.data$TRTACT <- factor(my.data$TRTACT, levels = unique(my.data$TRTACT[order(my.data$DOSE)]))
 
+#load dataset
+pkpd_data <- read.csv("../Data/Multiple_Ascending_Dose_Dataset2.csv")
+
+#ensure dataset has all the necessary columns
+  pkpd_data = pkpd_data %>%
+    mutate(ID      = ID,     #ID   column
+           TIME    = TIME,   #TIME column name 
+           NOMTIME = NOMTIME,#NOMINAL TIME column name
+           EVID    = EVID   ,#EVENT ID, >=1 is dose, otherwise measurement
+           LIDV    = LIDV,   #DEPENDENT VARIABLE column name
+           CENS    = CENS,   #CENSORING column name
+           CMT     = CMT,    #COMPARTMENT column
+           DOSE    = DOSE,   #DOSE column here (numeric value)
+           TRTACT  = TRTACT, #DOSE REGIMEN column here (character, with units),
+           LIDV_NORM = LIDV/DOSE,
+           LIDV_UNIT    = ifelse(CMT==2, "ng/ml",
+                                 ifelse(CMT==3, "IU/L", NA )
+                                 )
+    )
+#create a factor for the treatment variable for plotting
+pkpd_data = pkpd_data %>%
+  arrange(DOSE) %>%
+  mutate(TRTACT_low2high = factor(TRTACT, levels = unique(TRTACT)),
+         TRTACT_high2low = factor(TRTACT, levels = rev(unique(TRTACT))))
+
+#create pk and pd datasets
+pk_data <- pkpd_data %>%
+  filter(CMT==2)
+
+pd_data <- pkpd_data %>%
+  filter(CMT==3)
+
+#perform NCA, for additional plots
+NCA = pk_data %>%
+  group_by(ID, DOSE) %>%
+  filter(!is.na(LIDV)) %>%
+  summarize(AUC_last = caTools::trapz(TIME,LIDV),
```

Comment:
This is multiple dose, so replace AUC_last with AUC_0_24 and AUC_tau. See multiple dose PK Rmd, copied below:

            AUC_0_24 = caTools::trapz(TIME[NOMTIME > 0 & NOMTIME <= 24], LIDV[NOMTIME > 0 & NOMTIME <= 24]), 
            Cmax_0_24     = max(LIDV[NOMTIME>0 & NOMTIME<= 24]), 
            AUC_tau = caTools::trapz(TIME[NOMTIME > 120 & NOMTIME <= 144], LIDV[NOMTIME > 120 & NOMTIME <= 144]), 
            Cmax_tau     = max(LIDV[NOMTIME > 120 & NOMTIME <= 144]), 
            AUC_acc = AUC_tau / AUC_0_24, 
            Cmax_acc = Cmax_tau / Cmax_0_24